sql: failure in TestWindowFunctions

Found in an unrelated PR: https://teamcity.cockroachdb.com/viewLog.html?buildId=1824776&buildTypeId=Cockroach_UnitTests

```
=== RUN   TestWindowFunctions/spillForced=false/windowFunc:PERCENT_RANK_#02/batchSize=1/sel=false
panic: runtime error: index out of range [1] with length 0 [recovered]
	panic: runtime error: index out of range [1] with length 0

goroutine 8474 [running]:
testing.tRunner.func1(0xc000285400)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x29bdd60, 0xc0002e65e0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*percentRankWithPartitionOp).Next(0xc00041e000, 0x2f89da0, 0xc0000ea010, 0xc000940080, 0x5)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/relative_rank.eg.go:523 +0x11fc
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*simpleProjectOp).Next(0xc000113400, 0x2f89da0, 0xc0000ea010, 0xc000701200, 0x2ace8c2)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/simple_project.go:110 +0x62
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*opTestOutput).next(0xc0003da980, 0x2f89da0, 0xc0000ea010, 0x7f51ddd39d98, 0xc00002a600, 0x2ace8c2)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:904 +0xb5
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*opTestOutput).VerifyAnyOrder(0xc0003da980, 0xc0003da980, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:941 +0x74
github.com/cockroachdb/cockroach/pkg/sql/colexec.runTestsWithoutAllNullsInjection.func1(0xc000285400, 0xc00028abd0, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:325 +0x172
github.com/cockroachdb/cockroach/pkg/sql/colexec.runTestsWithFn.func1(0xc000285400)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:486 +0x294
testing.tRunner(0xc000285400, 0xc0001131d0)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0006eaf00, 0x2adf699, 0x13, 0x2b9aea8, 0x726801)
	/usr/local/go/src/testing/testing.go:961 +0x377
testing.runTests.func1(0xc000285100)
	/usr/local/go/src/testing/testing.go:1202 +0x78
testing.tRunner(0xc000285100, 0xc000677bf8)
	/usr/local/go/src/testing/testing.go:909 +0xc9
testing.runTests(0xc000258180, 0x4506580, 0x62, 0x62, 0x0)
	/usr/local/go/src/testing/testing.go:1200 +0x2a7
testing.(*M).Run(0xc000435c00, 0x0)
	/usr/local/go/src/testing/testing.go:1117 +0x176
github.com/cockroachdb/cockroach/pkg/sql/colexec.TestMain.func1(0xc000435c00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/main_test.go:66 +0x430
github.com/cockroachdb/cockroach/pkg/sql/colexec.TestMain(0xc000435c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/main_test.go:67 +0x30
main.main()
	_testmain.go:290 +0x135

goroutine 26 [syscall, 4 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:147 +0x9c
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:29 +0x41

goroutine 27 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:73 +0x73
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:45 +0x35

goroutine 28 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:100 +0x124
created by github.com/cockroachdb/cockroach/pkg/util/log.init.7
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:46 +0x4d

goroutine 52 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc0000c06c0)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:152 +0xdd
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:29 +0x57

goroutine 8321 [sync.Cond.Wait]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:310
sync.runtime_notifyListWait(0xc000701328, 0xc000000000)
	/usr/local/go/src/runtime/sema.go:510 +0xf8
sync.(*Cond).Wait(0xc000701318)
	/usr/local/go/src/sync/cond.go:56 +0x9d
github.com/cockroachdb/cockroach/pkg/storage.(*RocksDB).syncLoop(0xc000701200)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/rocksdb.go:678 +0x80
created by github.com/cockroachdb/cockroach/pkg/storage.(*RocksDB).open
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/rocksdb.go:665 +0x316

goroutine 8473 [chan receive]:
testing.(*T).Run(0xc000285400, 0xc0001e5a80, 0x15, 0xc0001131d0, 0x2)
	/usr/local/go/src/testing/testing.go:961 +0x377
github.com/cockroachdb/cockroach/pkg/sql/colexec.runTestsWithFn(0xc000285300, 0xc000343c60, 0x1, 0x1, 0x0, 0x0, 0x0, 0xc000718a80)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:467 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/colexec.runTestsWithoutAllNullsInjection(0xc000285300, 0xc000343c60, 0x1, 0x1, 0x0, 0x0, 0x0, 0xc0003722d0, 0xa, 0xa, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:319 +0x13d
github.com/cockroachdb/cockroach/pkg/sql/colexec.runTestsWithTyps(0xc000285300, 0xc000343c60, 0x1, 0x1, 0x0, 0x0, 0x0, 0xc0003722d0, 0xa, 0xa, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:207 +0xcb
github.com/cockroachdb/cockroach/pkg/sql/colexec.runTests(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/utils_test.go:191
github.com/cockroachdb/cockroach/pkg/sql/colexec.TestWindowFunctions.func1(0xc000285300)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/window_functions_test.go:274 +0x244
testing.tRunner(0xc000285300, 0xc00014cf00)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350

goroutine 8305 [chan receive]:
testing.(*T).Run(0xc000285300, 0xc000a93d40, 0x2a, 0xc00014cf00, 0x2)
	/usr/local/go/src/testing/testing.go:961 +0x377
github.com/cockroachdb/cockroach/pkg/sql/colexec.TestWindowFunctions(0xc0006eaf00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/window_functions_test.go:272 +0x6b3
testing.tRunner(0xc0006eaf00, 0x2b9aea8)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
FAIL	github.com/cockroachdb/cockroach/pkg/sql/colexec	254.018s
```

cc @yuzefovich for triage

--- a/pkg/sql/colexec/relative_rank_tmpl.go
+++ b/pkg/sql/colexec/relative_rank_tmpl.go
@@ -430,6 +430,9 @@ func (r *_RELATIVE_RANK_STRINGOp) Next(ctx context.Context) coldata.Batch {
 			// {{if .HasPartition}}
 			partitionCol := batch.ColVec(r.partitionColIdx).Bool()
 			var runningPartitionsSizesCol []int64
+			if r.partitionsState.runningSizes != nil {
+				runningPartitionsSizesCol = r.partitionsState.runningSizes.ColVec(0).Int64()
+			}
 			if sel != nil {
 				for _, i := range sel[:n] {
 					_COMPUTE_PARTITIONS_SIZES()
@@ -448,6 +451,9 @@ func (r *_RELATIVE_RANK_STRINGOp) Next(ctx context.Context) coldata.Batch {
 			// Next, we need to update the sizes of the peer groups.
 			peersCol := batch.ColVec(r.peersColIdx).Bool()
 			var runningPeerGroupsSizesCol []int64
+			if r.peerGroupsState.runningSizes != nil {
+				runningPeerGroupsSizesCol = r.peerGroupsState.runningSizes.ColVec(0).Int64()
+			}
 			if sel != nil {
 				for _, i := range sel[:n] {
 					_COMPUTE_PEER_GROUPS_SIZES()

